src/write.c

```diff
@@ -1200,11 +1213,11 @@ avifResult avifEncoderFinish(avifEncoder * encoder, avifRWData * output)
             }
         }
 
-        uint32_t imageWidth = imageMetadata->width;
-        uint32_t imageHeight = imageMetadata->height;
+        uint32_t imageWidth = cellWidth;
+        uint32_t imageHeight = cellHeight;
         if (isGrid) {
-            imageWidth = imageMetadata->width * item->gridCols;
-            imageHeight = imageMetadata->height * item->gridRows;
+            imageWidth = imageWidth * item->gridCols;
+            imageHeight = imageHeight * item->gridRows;
```

Comment:
Optional: Should we use the `*=` notation?

```
            imageWidth *=  item->gridCols;
            imageHeight *= item->gridRows;
```

Reply:
This looks better. Done.